cmake/CMakeLists.txt

```diff
@@ -70,22 +77,7 @@ if (${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU")
   option(ENABLE_COVERAGE "Enable code coverage" OFF)
   mark_as_advanced(ENABLE_COVERAGE)
   if(ENABLE_COVERAGE)
-    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --coverage")
-  endif()
-  option(ENABLE_SANITIZE_ADDRESS "Enable address sanitizer" OFF)
-  mark_as_advanced(ENABLE_SANITIZE_ADDRESS)
-  if(ENABLE_SANITIZE_ADDRESS)
-    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address")
-  endif()
-  option(ENABLE_SANITIZE_UNDEFINED "Enable undefined behavior sanitizer" OFF)
-  mark_as_advanced(ENABLE_SANITIZE_UNDEFINED)
-  if(ENABLE_SANITIZE_UNDEFINED)
-    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=undefined")
-  endif()
-  option(ENABLE_SANITIZE_THREAD "Enable thread sanitizer" OFF)
-  mark_as_advanced(ENABLE_SANITIZE_THREAD)
-  if(ENABLE_SANITIZE_THREAD)
-    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=thread")
+    set (CMAKE_TUNE_FLAGS "--coverage")
```

Comment:
So `ENABLE_COVERAGE` overwrites `CMAKE_TUNE_FLAGS`? If so we need:
```
CMAKE_DEPENDENT_OPTION(ENABLE_COVERAGE "Enable code coverage" OFF "NOT CMAKE_TUNE_FLAGS" OFF)
```

Reply:
That is kept only temporarily since coverage is used during regression testing. If this PR goes in, I can change the regression testing scripts to use the new flag/option and not use this anymore and then submit another PR to remove this option. Without any disruption or bogus regression failures. @rbberger is very busy right now and I don't want to distract him needlessly.

Reply:
If we remove that later, let's not worry about depend options for now.

Reply:
on closer inspection, I have reverted this section to the original. This is tied to testing and also triggers additional make targets. Folding it into CMAKE_TUNE_FLAGS would be more effort for little gain. So this will remain as is for now.